pages/login/LoginHandler.inc.php

```diff
@@ -166,104 +170,148 @@ function requestResetPassword($args, $request) {
-			$templateMgr->assign('error', 'user.login.lostPassword.invalidUser');
-			$templateMgr->display('frontend/pages/userLostPassword.tpl');
+			$templateMgr
+				->assign('error', 'user.login.lostPassword.invalidUser')
+				->display('frontend/pages/userLostPassword.tpl');
 
-		} else {
-			// Send email confirming password reset
-			import('lib.pkp.classes.mail.MailTemplate');
-			$mail = new MailTemplate('PASSWORD_RESET_CONFIRM');
-			$site = $request->getSite();
-			$this->_setMailFrom($request, $mail, $site);
-			$mail->assignParams([
-				'url' => $request->url(null, 'login', 'resetPassword', $user->getUsername(), array('confirm' => $hash)),
-				'siteTitle' => htmlspecialchars($site->getLocalizedTitle()),
-			]);
-			$mail->addRecipient($user->getEmail(), $user->getFullName());
-			$mail->send();
+			return;
+		}
 
-			$templateMgr->assign(array(
-				'pageTitle' => 'user.login.resetPassword',
-				'message' => 'user.login.lostPassword.confirmationSent',
-				'backLink' => $request->url(null, $request->getRequestedPage()),
-				'backLinkLabel' => 'user.login',
-			));
-			$templateMgr->display('frontend/pages/message.tpl');
+		if ($user->getDisabled()) {
+			$templateMgr
+				->assign([
+					'error' => 'user.login.lostPassword.confirmationSentFailedWithReason',
+					'reason' => empty($reason = $user->getDisabledReason() ?? '')
+						? __('user.login.accountDisabled')
+						: __('user.login.accountDisabledWithReason', ['reason' => $reason])
+				])
+				->display('frontend/pages/userLostPassword.tpl');
+			
+			return;
 		}
+
+		// Send email confirming password reset as all check has passed
+		import('lib.pkp.classes.mail.MailTemplate');
+		$mail = new MailTemplate('PASSWORD_RESET_CONFIRM');
+		$site = $request->getSite();
+		$this->_setMailFrom($request, $mail, $site);
+		$mail->assignParams([
+			'url' => $request->url(null, 'login', 'resetPassword', $user->getUsername(), array('confirm' => $hash)),
+			'siteTitle' => htmlspecialchars($site->getLocalizedTitle()),
+		]);
+		$mail->addRecipient($user->getEmail(), $user->getFullName());
+		$mail->send();
+
+		$templateMgr->assign([
+			'pageTitle' => 'user.login.resetPassword',
+			'message' => 'user.login.lostPassword.confirmationSent',
+			'backLink' => $request->url(null, $request->getRequestedPage()),
+			'backLinkLabel' => 'user.login',
+		]);
+		$templateMgr->display('frontend/pages/message.tpl');
 	}
 
 	/**
-	 * Reset a user's password
+	 * Present the password reset form to reset user's password
 	 * @param $args array first param contains the username of the user whose password is to be reset
 	 */
 	function resetPassword($args, $request) {
+
+		if (Validation::isLoggedIn()) {
+			$this->sendHome($request);
+		}
+		
+		$this->_isBackendPage = true;
 		$this->setupTemplate($request);
+		$templateMgr = TemplateManager::getManager($request);
+		$templateMgr->setupBackendPage();
+		$templateMgr->assign([
+			'pageTitle' => __('user.changePassword'),
```

Comment:
This causes a double-translated locale key:
![image](https://user-images.githubusercontent.com/200411/212782201-7033daad-90a2-4909-b06e-804475f224ce.png)
The `pageTitle` variable is expected to be an untranslated locale key, not a translated one; see:

https://github.com/pkp/pkp-lib/blob/5419ce37628ae0d922db6ecf1a31ae7f6bf79640/templates/frontend/pages/message.tpl#L16

You can fix this by removing the `__( ... )` wrapper around `'user.changePassword'`.

Reply:
Actually, I attached this to the wrong line of code. This `pageTitle` is never used and can be removed, I think; I'll link the other one to the above comment.

Reply:
Fixed .